code/game/objects/effects/decals/contraband.dm

```diff
@@ -14,16 +14,23 @@
 	icon_state = "rolled_poster"
 	drop_sound = 'sound/items/drop/wrapper.ogg'
 	pickup_sound = 'sound/items/pickup/wrapper.ogg'
-	var/serial_number = 0
-
-
-/obj/item/contraband/poster/Initialize(mapload, given_serial = 0)
+	/// Set this variable to a /singleton/poster_design specify a desired poster.
+	/// If unset, is randomly set on `Initialize()`
+	var/singleton/poster_design/poster_type
+	/// If `poster_type` is unset, this is used to select a new design through stricter categories
+	/// See `poster_subtypes.dm` for implementations
+	var/random_type = /singleton/poster_design
+
+/obj/item/contraband/poster/Initialize(mapload, given_type)
 	. = ..()
-	if(given_serial == 0)
-		serial_number = rand(1, GLOB.poster_designs.len)
+	if(!poster_type)
+		if(!given_type)
+			poster_type = pick(GET_SINGLETON_SUBTYPE_LIST(random_type))
+		else
+			poster_type = given_type
```

Comment:
random posters should be done via `/obj/random` (see potted plants or other things for a example)